tests/issues/1163_test.go

```diff
@@ -0,0 +1,43 @@
+package issues
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	clickhouse_tests "github.com/ClickHouse/clickhouse-go/v2/tests"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIssue1163(t *testing.T) {
+	env, err := GetIssuesTestEnvironment()
+	require.NoError(t, err)
+	useSSL, err := strconv.ParseBool(clickhouse_tests.GetEnv("CLICKHOUSE_USE_SSL", "false"))
+	require.NoError(t, err)
+	var tlsConfig *tls.Config
+	port := env.Port
+	if useSSL {
+		tlsConfig = &tls.Config{}
+		port = env.SslPort
+	}
+	options := &clickhouse.Options{
+		Addr:  []string{fmt.Sprintf("%s:%d", env.Host, port)},
+		Debug: true,
+		Debugf: func(format string, v ...any) {
+
```

Comment:
it's worth adding a flag here, so it's assert later
```suggestion
debugfCalled = true
```

Reply:
thanks for reviewing. Addressed MR comment